airbyte-test-utils/src/main/java/io/airbyte/test/utils/AirbyteAcceptanceTestHelper.java

```diff
@@ -0,0 +1,721 @@
+
+  private static final String DOCKER_COMPOSE_FILE_NAME = "docker-compose.yaml";
+  // assume env file is one directory level up from airbyte-tests.
+  private final static File ENV_FILE = Path.of(System.getProperty("user.dir")).getParent().resolve(".env").toFile();
+
+  private static final String SOURCE_E2E_TEST_CONNECTOR_VERSION = "0.1.1";
+  private static final String DESTINATION_E2E_TEST_CONNECTOR_VERSION = "0.1.1";
+
+  private static final Charset UTF8 = StandardCharsets.UTF_8;
+  private static final boolean IS_KUBE = System.getenv().containsKey("KUBE");
+  private static final boolean IS_MINIKUBE = System.getenv().containsKey("IS_MINIKUBE");
+  private static final boolean IS_GKE = System.getenv().containsKey("IS_GKE");
+  private static final boolean IS_MAC = System.getProperty("os.name").startsWith("Mac");
+  private static final boolean USE_EXTERNAL_DEPLOYMENT =
+      System.getenv("USE_EXTERNAL_DEPLOYMENT") != null && System.getenv("USE_EXTERNAL_DEPLOYMENT").equalsIgnoreCase("true");
+
+  private static final String OUTPUT_NAMESPACE_PREFIX = "output_namespace_";
+  private static final String OUTPUT_NAMESPACE = OUTPUT_NAMESPACE_PREFIX + "${SOURCE_NAMESPACE}";
+  private static final String OUTPUT_STREAM_PREFIX = "output_table_";
+  private static final String TABLE_NAME = "id_and_name";
+  public static final String STREAM_NAME = TABLE_NAME;
+  public static final String COLUMN_ID = "id";
+  public static final String COLUMN_NAME = "name";
+  private static final String COLUMN_NAME_DATA = "_airbyte_data";
+  private static final String SOURCE_USERNAME = "sourceusername";
+  public static final String SOURCE_PASSWORD = "hunter2";
+
+  /**
+   * When the acceptance tests are run against a local instance of docker-compose or KUBE then these
+   * test containers are used. When we run these tests in GKE, we spawn a source and destination
+   * postgres database ane use them for testing.
+   */
+  private PostgreSQLContainer sourcePsql;
+  private PostgreSQLContainer destinationPsql;
+  private AirbyteTestContainer airbyteTestContainer;
+  private AirbyteApiClient apiClient;
+  private UUID workspaceId;
+
+  private KubernetesClient kubernetesClient = null;
+
+  private List<UUID> sourceIds;
+  private List<UUID> connectionIds;
+  private List<UUID> destinationIds;
+  private List<UUID> operationIds;
+
+  public PostgreSQLContainer getSourcePsql() {
+    return sourcePsql;
+  }
+
+  public KubernetesClient getKubernetesClient() {
+    return kubernetesClient;
+  }
+
+  public void removeConnection(final UUID connection) {
+    connectionIds.remove(connection);
+  }
+
+  public void setApiClient(final AirbyteApiClient apiClient) {
+    this.apiClient = apiClient;
+  }
+
+  @SuppressWarnings("UnstableApiUsage")
+  public void init(final AirbyteApiClient apiClient) throws URISyntaxException, IOException, InterruptedException, ApiException {
```

Comment:
two thoughts here:
1) instead of having an explicit init, would it be simpler to bundle this within the constructor and only call the constructor in each test's `setup` function?
2) less important, this function seems long.

Reply:
e.g. instead of creating this https://github.com/airbytehq/airbyte/pull/13950/files#diff-889e4242c3b54993a63d9fed4a1e1e82253ae3020c9c636f3f1082a43e16f709R94 and having to call `init`, we only creating the test helper class once.

Reply:
i don't think the link is working as intended, so i'm not sure if i understand your second comment. The `init` function should be only called once in the `BeforeAll` section.

Regarding the first comment:
1. Makes sense to use a constructor instead of an init function. Although i think the constructor should only be called once in the `BeforeAll` section, instead of `BeforeEach`
2. feel like i can move out the logging of connectors being used into the acceptance test files themselves, doesn't make it that much shorter though.
```
    // log which connectors are being used.
    final SourceDefinitionRead sourceDef = apiClient.getSourceDefinitionApi()
        .getSourceDefinition(new SourceDefinitionIdRequestBody()
            .sourceDefinitionId(UUID.fromString("decd338e-5647-4c0b-adf4-da0e75f5a750")));
    final DestinationDefinitionRead destinationDef = apiClient.getDestinationDefinitionApi()
        .getDestinationDefinition(new DestinationDefinitionIdRequestBody()
            .destinationDefinitionId(UUID.fromString("25c5221d-dce2-4163-ade9-739ef790f503")));
    LOGGER.info("pg source definition: {}", sourceDef.getDockerImageTag());
    LOGGER.info("pg destination definition: {}", destinationDef.getDockerImageTag());
    ```

Reply:
> called once in the BeforeAll section, instead of BeforeEach

yeap! when I said 'each test' I was referring to each acceptance test.

Reply:
> feel like i can move out the logging of connectors being used into the acceptance test files themselves, doesn't make it that much shorter though.

agree - this was more of a 'code smell' comment rather than a concrete suggestion. one thing I can think of is breaking this into smaller private function centered around the various deployment modes we have here e.g. 1) Kube 2) external deployment etc. this is a nit since majority of the benefit is pulling all of these out of the base acceptance tests classes.